### DISCUSSION TITLE
$expand=principal working differently from Graph Explorer?

### DISCUSSION THREAD

**Discussion by TheodorNygard** [microsoftgraph/msgraph-sdk-powershell#2364] [Q&A] [answered]
Good day!

Hoping I'm just missing something fundamental here,

Querying the following in Graph Explorer gets me rich output on the specific user assigned to the role, prime for some $select etc.
`https://graph.microsoft.com/beta/roleManagement/directory/roleAssignments?$filter=roleDefinitionId eq '62e90394-69f5-4237-9190-012177145e10'&$expand=principal`

Attempting to run the same through Powershell and msgraph-sdk and it doesn't appear to actually expand. running the below with or without the $expand returns the same content - specifically I get the roleassignment but no further.
`Invoke-MgGraphRequest -method GET -Uri "https://graph.microsoft.com/beta/roleManagement/directory/roleAssignments?$filter=roleDefinitionId eq '62e90394-69f5-4237-9190-012177145e10'&$expand=principal" -OutputType PSObject`

**Comment by salbeck-sit** [comment, marked as answer]
you probably need to escape the $-signs in the Uri ($filter and $expand) or use single quotes

**Reply by TheodorNygard** [reply]
Absolutely right on the money! Thanks
Something fundamental it was then, if anyone finds this - it won't throw an error if you're missing it, but the ol' backtick will give you the correct syntax: `$var
